mentoring/components/mcq.py

```diff
@@ -0,0 +1,140 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2014 Harvard
+#
+# Authors:
+#          Xavier Antoviaque <xavier@antoviaque.org>
+#
+# This software's license gives you freedom; you can copy, convey,
+# propagate, redistribute and/or modify this program under the terms of
+# the GNU Affero General Public License (AGPL) as published by the Free
+# Software Foundation (FSF), either version 3 of the License, or (at your
+# option) any later version of the AGPL published by the FSF.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU Affero
+# General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program in a file in the toplevel directory called
+# "AGPLv3".  If not, see <http://www.gnu.org/licenses/>.
+#
+
+# Imports ###########################################################
+
+import logging
+
+from xblock.fields import Scope, String, List
+from xblock.fragment import Fragment
+from xblockutils.resources import ResourceLoader
+
+from .questionnaire import QuestionnaireAbstractBlock
+
+
+# Globals ###########################################################
+
+log = logging.getLogger(__name__)
+loader = ResourceLoader(__name__)
+
+
+# Classes ###########################################################
+
+class MCQBlock(QuestionnaireAbstractBlock):
+    """
+    An XBlock used to ask multiple-choice questions
+    """
+    student_choice = String(help="Last input submitted by the student", default="", scope=Scope.user_state)
+
+    correct_choices = List(
+        display_name="Correct Choice[s]",
+        help="Enter the value[s] that students may select for this question to be considered correct. ",
+        scope=Scope.content,
+        list_editor="comma-separated",
+    )
+    editable_fields = QuestionnaireAbstractBlock.editable_fields + ('correct_choices', )
+
+    def describe_choice_correctness(self, choice_value):
+        if choice_value in self.correct_choices:
+            if len(self.correct_choices) == 1:
+                return u"Correct"
+            return u"Acceptable"
+        else:
+            if len(self.correct_choices) == 1:
+                return u"Wrong"
+            return u"Not Acceptable"
```

Comment:
The solution to the MCQ/MRQs appear in the title : )

![screenshot-sandbox opencraft com 2015-02-24 16-58-02](https://cloud.githubusercontent.com/assets/514483/6352953/8338692e-bc46-11e4-9a54-16357c49c186.png)